spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -357,6 +362,28 @@ public void setBatchReceiveTimeout(long batchReceiveTimeout) {
 		this.batchReceiveTimeout = batchReceiveTimeout;
 	}
 
+	/**
+	 * Set the interval for checking for pending replies during shutdown.
+	 * Default is 200ms.
+	 * @param pendingReplyCheckInterval the interval in milliseconds.
+	 * @since 4.0
+	 */
+	public void setPendingReplyCheckInterval(long pendingReplyCheckInterval) {
+		this.pendingReplyCheckInterval = pendingReplyCheckInterval;
+	}
+
+	/**
+	 * Set a provider for the number of pending replies.
+	 * When set, the container will wait for pending replies during shutdown,
+	 * up to the {@link #setShutdownTimeout(long) shutdownTimeout}.
+	 * @param pendingReplyProvider the pending reply provider.
+	 * @since 4.0
+	 * @see org.springframework.amqp.rabbit.core.RabbitTemplate#getPendingReplyCount()
+	 */
+	public void setPendingReplyProvider(PendingReplyProvider pendingReplyProvider) {
```

Comment:
I don't think we need this extra abstraction.
We already have a `ListenerContainerAware` implemented on the `RabbitTemplate`.
So, that `getPendingReplyCount()` could be moved into that contract.